From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Dohrmann <erbse.13@gmx.de>
Date: Mon, 23 Sep 2024 09:26:32 +0200
Subject: [PATCH] i386: omit some unneeded ACPI tables

This makes the generated ACPI code more deterministic and less dependent on the
guest configuration (e.g. the amount of assigned memory).
---
 hw/i386/acpi-build.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/hw/i386/acpi-build.c b/hw/i386/acpi-build.c
index 423c4959fe809b76f86814997fce47dbea3572b9..f99e87c29b6669ff7a06c3602667a6fcdbb930bd 100644
--- a/hw/i386/acpi-build.c
+++ b/hw/i386/acpi-build.c
@@ -1063,6 +1063,8 @@ build_dsdt(GArray *table_data, BIOSLinker *linker,
                          mcfg.base, mcfg.base + mcfg.size - 1);
     }
 
+#if 0
+
     scope = aml_scope("\\_SB.PCI0");
     /* build PCI0._CRS */
     crs = aml_resource_template();
@@ -1127,6 +1129,8 @@ build_dsdt(GArray *table_data, BIOSLinker *linker,
 #endif
     aml_append(scope, aml_name_decl("_CRS", crs));
 
+#endif
+
     /* reserve GPE0 block resources */
     dev = aml_device("GPE0");
     aml_append(dev, aml_name_decl("_HID", aml_string("PNP0A06")));
@@ -2028,6 +2032,7 @@ void acpi_build(AcpiBuildTables *tables, MachineState *machine)
         }
     }
 #endif
+#if 0
     if (machine->numa_state->num_nodes) {
         acpi_add_table(table_offsets, tables_blob);
         build_srat(tables_blob, tables->linker, machine);
@@ -2042,6 +2047,7 @@ void acpi_build(AcpiBuildTables *tables, MachineState *machine)
                        x86ms->oem_id, x86ms->oem_table_id);
         }
     }
+#endif
     if (acpi_get_mcfg(&mcfg)) {
         acpi_add_table(table_offsets, tables_blob);
         build_mcfg(tables_blob, tables->linker, &mcfg, x86ms->oem_id,
